components/script/dom/window.rs

```diff
@@ -989,6 +987,51 @@ impl Window {
         global_scope.constellation_chan().send(message).unwrap();
     }
 
+    /// https://html.spec.whatwg.org/multipage/#scroll-to-the-fragment-identifier
+    pub fn check_and_scroll_fragment(&self, fragment: &str) {
+        let doc = self.Document();
+        let body = doc.GetBody();
+        let target = doc.find_fragment_node(fragment);
+
+        // Step 1
+        doc.set_target_element(target.r().map(|e| &*e));
+
+        let target = target.r().map(|e| &*e).or_else(
+            // Step 2
+            || if fragment == "" {
+                body.r().map(|b| b.upcast::<Element>())
```

Comment:
This looks wrong. When `fragment` is the empty string, the `find_fragment_node` algorithm points to the document node instead of the document's body.